go-controller/pkg/ovn/pods.go

```diff
@@ -613,6 +610,27 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) (err error) {
 	return nil
 }
 
+func (oc *Controller) updatePodAnnotationWithRetry(origPod *kapi.Pod, podInfo *util.PodAnnotation) error {
+	resultErr := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		// Informer cache should not be mutated, so get a copy of the object
+		pod, err := oc.kube.GetPod(origPod.Namespace, origPod.Name)
```

Comment:
Isn't this getting the pod from the API server ? 

Are you getting the pod from the API server in case your attempt to mutate the pod (add the annotations) failed, and for whatever reason the informer was not updated yet ?

Reply:
Yes, the informer cache won't update, without this, if there is any conflict, the pod annotation update would fail.

Reply:
Wouldn't it be preferable ? If would avoid overloading the API.

AFAIU, if there's a conflict, it's because someone else manged to update the pod first - and those changes will trickle down to the pod eventually (I am hoping soon).

Anyway, my main goal was to make sure I understood the reason why. Thanks for explaining.

Reply:
In theory we could use watcher.GetPod() instead. whenever there is an update, the watcher should get that update and update its local cache. Not sure how fast it is though. I am a little nervous as the current code has been through quite some testing/product usage and we know it performance impact, whereas the other does not. please suggest...

Reply:
the current code handles the update in watchPodHander to handle the update. Not sure if this is your question. In our code, we'd need to get up-to-date pod information in order to avoid another retryConflict, kube.GetPod() would make sure of that by directly get it from the api-server; watcher.GetPod() would depends on if watcher has already get update from api-server and updated the local cache.

Reply:
Still not using the informer here to get the pod?

Reply:
it has changed to use oc.watchFactory.GetPod